### PR TITLE
AB#478509 Use BetterType instead of String for Conformances so we have a namespace for conformances

### PR DESCRIPTION
**Pull request by dvosscricut**
AB#478509 Use BetterType instead of String for Conformances so we have a namespace for conformances

### CODE REVIEW

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift [hunk 1]
```diff
@@ -886,8 +888,8 @@ struct TranslatedProtocol: TranslatedType {
                 constructor: .reference,
                 fields: externalWitnessFields,
                 methods: externalWitnessMethods,
-                conformances: ["com.cricut.fishyjoes.runtime.SwiftReference(_swiftReference)"]
-            ).conforming(to: [sourceType.name], context: context)
+                conformances: [BetterType.named(.init(name: "SwiftReference(_swiftReference)", module: "fishyjoes.runtime"))]
```
Comment (cobbal) [inline Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift]: This is the name of a kotlin class, It should be either a `KType` or a string, BetterType refers to swift types. (Maybe it should be called SwiftType... I'll have to think about that)

File: Sources/FishyJoesCore/Unparse/CSharpClass.swift [hunk 2]
```diff
@@ -42,22 +42,22 @@ class CSharpClass: NestedClass {
     var innerClasses: [CSharpClass] = []
     let fields: [Variable]
     let methods: [Method]
-    let conformances: [String]
+    let conformances: Set<BetterType>
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/CSharpClass.swift]: Same here

**Review by cobbal — CHANGES_REQUESTED** [review]
Nice. That worked out well. There are just a couple places where a swift type is placed into a context that should have a different-lannguage type.

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift (hunk 1 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift]: updated to use KotlinClass.KType inside The Kotlin Kingdom

We still use BetterType/SType when dwelling within the borders of the Swift Sovereign State.

File: Sources/FishyJoesCore/Unparse/CSharpClass.swift (hunk 2 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/Unparse/CSharpClass.swift]: CSharpClass.CSType is now used within the C# Kingdom

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift [hunk 3]
```diff
@@ -108,6 +108,10 @@ extension TranslatedEnum {
 
         let (enumFields, enumMethods) = CSharpClass.separate(fieldsAndMethods: fieldsAndMethods)
 
+        let csTypeConformances = conformances.map {
+            CSharpClass.CSType.named(package: $0.module, name: $0.nonNamespacedName)
```
Comment (cobbal) [inline Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift]: This should be resolved and the csharptype pulled from the translatedtype. The swift type may not be named the same thing.

**Review by cobbal — CHANGES_REQUESTED** [review]

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift (hunk 3 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift]: ok! <high pitched lemming voice>

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift [hunk 4]
```diff
@@ -109,7 +109,8 @@ extension TranslatedEnum {
         let (enumFields, enumMethods) = CSharpClass.separate(fieldsAndMethods: fieldsAndMethods)
 
         let csTypeConformances = conformances.map {
-            CSharpClass.CSType.named(package: $0.module, name: $0.nonNamespacedName)
+            let resolved = context.resolve(type: $0)
+            return CSharpClass.CSType.named(package: resolved.sourceType.module, name: resolved.sourceType.nonNamespacedName)
```
Comment (cobbal) [inline Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift]: Use this: https://github.com/cricut/FishyJoes/blob/main/Sources/FishyJoesCore/TranslatedTypes/TranslatedType.swift#L11

Comment (dvosscricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedEnum+cSharp.swift]: ok, I did it for kotlinType as well

**Review by cobbal — APPROVED** [review]
Looks very good

**Pull request merged**